src/main/java/roomescape/reservation/ReservationService.java

```diff
@@ -1,30 +1,62 @@
 package roomescape.reservation;
 
 import org.springframework.stereotype.Service;
+import roomescape.theme.ThemeRepository;
+import roomescape.time.TimeRepository;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
 
 @Service
 public class ReservationService {
-    private ReservationDao reservationDao;
-
-    public ReservationService(ReservationDao reservationDao) {
-        this.reservationDao = reservationDao;
+    private final ReservationRepository reservationRepository;
+    private final TimeRepository timeRepository;
+    private final ThemeRepository themeRepository;
+
+    public ReservationService(
+            ReservationRepository reservationRepository,
+            TimeRepository timeRepository,
+            ThemeRepository themeRepository
+    ) {
+        this.reservationRepository = reservationRepository;
+        this.timeRepository = timeRepository;
+        this.themeRepository = themeRepository;
     }
 
     public ReservationResponse save(ReservationRequest reservationRequest) {
-        Reservation reservation = reservationDao.save(reservationRequest);
+        Reservation reservation = reservationRepository.save(requestToDao(reservationRequest));
 
         return new ReservationResponse(reservation.getId(), reservationRequest.name(), reservation.getTheme().getName(), reservation.getDate(), reservation.getTime().getValue());
     }
 
+    private Reservation requestToDao(ReservationRequest request) {
+        return new Reservation(request.name(), request.date(), timeRepository.findById(request.time()).get(), themeRepository.findById(request.theme()).get());
+
+    }
+
     public void deleteById(Long id) {
-        reservationDao.deleteById(id);
+        reservationRepository.deleteById(id);
     }
 
     public List<ReservationResponse> findAll() {
-        return reservationDao.findAll().stream()
+        return StreamSupport.stream(reservationRepository.findAll().spliterator(), false)
+                .toList()
+                .stream()
                 .map(it -> new ReservationResponse(it.getId(), it.getName(), it.getTheme().getName(), it.getDate(), it.getTime().getValue()))
```

Comment:
```suggestion
                .map(it -> new ReservationResponse(
                it.getId(), 
                it.getName(), 
                it.getTheme().getName(), 
                it.getDate(), 
                it.getTime().getValue()))
```
코드의 길이가 길고 컬럼을 3개 이상 받고 있으니 이렇게 바꾸면 가독성이 더 좋아질 것으로 보여요!